dart/lib/src/enricher/web_enricher_event_processor.dart

```diff
@@ -0,0 +1,102 @@
+import 'dart:async';
+
+import '../platform_checker.dart';
+
+import '../protocol.dart';
+import 'dart:html' as html show window, Window;
+
+import '../sentry_options.dart';
+
+EventProcessor enricherEventProcessor(SentryOptions options) {
+  final processor = WebEnricherEventProcessor(
+    html.window,
+    options.platformChecker,
+  );
+  return processor.apply;
+}
+
+class WebEnricherEventProcessor {
+  WebEnricherEventProcessor(
+    this._window,
+    this._platformChecker,
+  );
+
+  final html.Window _window;
+  final PlatformChecker _platformChecker;
+
+  FutureOr<SentryEvent> apply(SentryEvent event, {dynamic hint}) async {
+    // Web has no native integration, so no need to check for it
+
+    final contexts = event.contexts.copyWith(
+      device: await _getDevice(event.contexts.device),
+      operatingSystem: _getOperatingSystem(event.contexts.operatingSystem),
+      runtimes: _getRuntimes(event.contexts.runtimes),
+    );
+
+    return event.copyWith(
+      contexts: contexts,
+    );
+  }
+
+  Future<SentryDevice> _getDevice(SentryDevice? device) async {
+    return (device ?? SentryDevice()).copyWith(
+      online: device?.online ?? _window.navigator.onLine,
+      memorySize: device?.memorySize ?? _getMemorySize(),
+      orientation: device?.orientation ?? _getScreenOrientation(),
+      screenHeightPixels: device?.screenHeightPixels ??
+          _window.screen?.available.height.toInt(),
+      screenWidthPixels:
+          device?.screenWidthPixels ?? _window.screen?.available.width.toInt(),
+      screenDensity:
+          device?.screenDensity ?? _window.devicePixelRatio.toDouble(),
+      timezone: device?.timezone ?? DateTime.now().timeZoneName,
+    );
+  }
+
+  SentryOperatingSystem _getOperatingSystem(SentryOperatingSystem? os) {
+    return (os ?? SentryOperatingSystem()).copyWith(
+      name: os?.name ?? _platformChecker.platform.operatingSystem,
+    );
+  }
+
+  List<SentryRuntime> _getRuntimes(List<SentryRuntime>? runtimes) {
+    // Pure Dart doesn't have specific runtimes per build mode
+    // like Flutter: https://flutter.dev/docs/testing/build-modes
+    final dartRuntime = SentryRuntime(
+      key: 'sentry_dart_runtime',
+      name: 'Dart',
+    );
```

Comment:
wondering if its useful if `Platform.version` does not exist here

Reply:
Your call, I just included it because all other have it too.

Reply:
mm, yeah I just don't see the benefit for hardcoded values, I'd leave the comment for future reference but the `dartRuntime` would be gone as it does not say anything